### PR TITLE
Updates for tox 4

### PR DESCRIPTION
**Pull request by aliciaaevans**
Configuration updates for tox 4 and consolidation of some fields.

Using package=editable-legacy instead of usedevelop=true to avoid errors due to new packaging behavior: https://tox.wiki/en/latest/faq.html#tox-4-packaging-changes

Also removed some old pylint options that are no longer valid. Upgraded to py39 and MacOS 11.

Updated the hard-coded numbers that test that GO data was loaded. Also added comments that indicate where the information can be found in the future.

### CODE REVIEW

File: azure-template-tox-job.yml [hunk 1]
```diff
@@ -40,7 +42,7 @@ jobs:
       - bash: |
           set -e
           ls -la /Applications/Xcode*
-          sudo xcode-select --switch /Applications/Xcode_10.2.app/Contents/Developer
+          sudo xcode-select --switch /Applications/Xcode_12.5.1.app/Contents/Developer
```
Comment (piehld) [inline azure-template-tox-job.yml:45]: Thanks for updating this too

File: rcsb/utils/tests-go/testGeneOntologyProvider.py [hunk 2]
```diff
@@ -58,7 +58,10 @@ def testReloadGeneOntology(self):
             self.assertEqual(len(rL), 3)
 
             #
-            goIdL = [("GO:0023052", 1), ("GO:0051179", 1), ("GO:0070727", 4), ("GO:1990747", 30)]
+            # These numbers may change as ome terms become obsolete
+            # in future GO updates.
+            # See http://geneontology.org/stats.html and https://github.com/geneontology/go-announcements
```
Comment (piehld) [inline rcsb/utils/tests-go/testGeneOntologyProvider.py:63]: Thank you for looking into this!

**Review by piehld — APPROVED** [review]
LGTM--Thanks!

**Comment by piehld** [comment]
@aliciaaevans Sorry just remembered, maybe we should also add the regular testing trigger for every Sunday, along with the rest of the repos?

**Comment by aliciaaevans** [comment]
@piehld 
I looked in azure-pipelines.yml and the cron configuration is the same as the other repos. Is there somewhere else that is controlled?

**Comment by piehld** [comment]
@aliciaaevans I wonder if it's an indentation issue in the `azure-pipelines.yml` file? Seems like the indentation for that section is different in this repo versus others...

e.g., in `insilico3d`, this is the indentation:
```
schedules:
  - cron: "0 12 * * 0"
    displayName: Weekly Sunday build
    branches:
      include:
        - master
    always: true
```

**Comment by aliciaaevans** [comment]
@piehld Interesting, I was comparing it with py-rcsb_utils_io and it has the same indentation.

**Comment by piehld** [comment]
@aliciaaevans OK good point. I've just tried to change something in the Azure settings...let's see if that sticks this weekend...

**Comment by piehld** [comment]
BTW, for record's sake, I think we have to click "sync schedules" from the pipeline page (bottom right corner):
Screenshot 2023-01-12 at 9 48 27 AM

**Pull request merged**